api/api.proto

```diff
@@ -83,6 +91,7 @@ message HeartBeat {
 message StreamFinished {
   string WorkerID = 1;
   uint32 StreamID = 2;
+  bool EndedEarly = 3;
```

Comment:
Why does TUM Live need this info? Can't we always set a stream to "done" once `StreamFinished` was called?

Reply:
Actually, the logic should be somewhat the same. This information is used to allow waiting rooms to properly be displayed e.g., but even better if we also got this fix for self streams and so on.

I just was unsure in this regard: Lecture halls streams should never end early, Selfstreams can, but I was unsure what would happen if e.g. OBS crashed or restarted. Could this somehow trigger `StreamFinished`? If so, we'd need to set "not done" again, when the stream restarts.

Reply:
If the connection ungracefully ends, it can happen that `streamFinished` is called. Also, if there are problems with the stream (e.g. sound), lecturers often stop the stream and start it again, so we need to expect streams to end and start again.

Reply:
Okay, that makes sense. I added the necessary logic in the backend now: https://github.com/joschahenningsen/TUM-Live/pull/187